aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -135,15 +136,33 @@ Tensor angle(const Tensor& self) {
 }
 
 Tensor real(const Tensor& self) {
-  TORCH_CHECK(!self.is_complex(), "real is not yet implemented for complex tensors.");
-  return self;
+  if (self.is_complex()) {
+    auto float_tensor = at::native::view_complex_as_float(self);
+    if (self.numel() == 0) {
+      return float_tensor;
+    } else {
+      return at::select(float_tensor, float_tensor.dim() - 1, 0);
+    }
+  } else {
+    TORCH_CHECK(false, "real is not implemented for tensors with non-complex dtypes.");
+    // Note: unreachable
+    return self;
```

Comment:
I know this is not reachable but please make it consistent between real and imag.